src/libDirectoryService/FinalBlockPreProcessing.cpp

```diff
@@ -896,6 +896,7 @@ void DirectoryService::RunConsensusOnFinalBlock(bool revertStateDelta)
         SetState(FINALBLOCK_CONSENSUS);
         cv_finalBlockConsensusObject.notify_all();
     }
+    CommitFinalBlockConsensusBuffer();
```

Comment:
In this case the cv below to trigger view change may always times out, as the triggering happens in processing the consensus message but in the same thread. I think currently we should detach this CommitFinalBlockConsensusBuffer first in order to let the cv get notified to avoid initiating view change